File: server_integration_tests/kotlin/kotlin_client/client-app/src/test/kotlin/dev/graphlink/kotlinclienttest/MutationsTest.kt
package dev.graphlink.kotlinclienttest

import dev.graphlink.kotlinclient.generated.client.GraphLinkClient
import dev.graphlink.kotlinclient.generated.inputs.CreateArticleInput
import dev.graphlink.kotlinclient.generated.inputs.UpdateArticleInput
import kotlinx.coroutines.test.runTest
import kotlin.test.BeforeTest
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class MutationsTest {

    private lateinit var client: GraphLinkClient

    @BeforeTest
    fun setUp() {
        client = Fixtures.newClient()
    }

    @Test
    fun createArticle_returnsCreatedArticle() = runTest {
        val res = client.mutations.createArticle(CreateArticleInput(title = "New Post", authorId = "2"))
        assertEquals("New Post", res.createArticle.title)
        assertEquals("2", res.createArticle.authorId)
        assertTrue(res.createArticle.id.isNotEmpty())
    }

    @Test
    fun updateArticle_updatesTitle() = runTest {
        val created = client.mutations.createArticle(CreateArticleInput(title = "Original Title", authorId = "1"))
        val res = client.mutations.updateArticle(UpdateArticleInput(id = created.createArticle.id, title = "Updated Title"))
        assertEquals(created.createArticle.id, res.updateArticle.id)
        assertEquals("Updated Title", res.updateArticle.title)
    }

    @Test
    fun deleteArticle_returnsTrueWhenArticleExists() = runTest {
        val created = client.mutations.createArticle(CreateArticleInput(title = "To Delete", authorId = "1"))
        val res = client.mutations.deleteArticle(created.createArticle.id)
        assertTrue(res.deleteArticle)
    }

    @Test
    fun deleteArticle_returnsFalseWhenArticleDoesNotExist() = runTest {
        val res = client.mutations.deleteArticle("missing-id")
        assertFalse(res.deleteArticle)
    }
}

File: server_integration_tests/clients/swift_client/Tests/GraphLinkGeneratedTests/MutationsTest.swift
import XCTest
@testable import GraphLinkGenerated

final class MutationsTest: XCTestCase {
    var client: GraphLinkClient!

    override func setUp() {
        client = Fixtures.newClient()
    }

    func testCreateArticle_returnsCreatedArticleWithGivenInput() async throws {
        let res = try await client.mutations.createArticle(
            input: CreateArticleInput(title: "New Post", authorId: "2"),
            fieldArgs: CreateArticleFieldArgs(latestArticlesLimit: 2)
        )
        XCTAssertEqual(res.createArticle.title, "New Post")
        XCTAssertEqual(res.createArticle.authorId, "2")
        XCTAssertFalse(res.createArticle.id.isEmpty)
    }

    func testUpdateArticle_updatesTitleAndReturnsArticle() async throws {
        let created = try await client.mutations.createArticle(
            input: CreateArticleInput(title: "Original Title", authorId: "1"),
            fieldArgs: CreateArticleFieldArgs(latestArticlesLimit: 2)
        )
        let res = try await client.mutations.updateArticle(
            input: UpdateArticleInput(id: created.createArticle.id, title: "Updated Title"),
            fieldArgs: UpdateArticleFieldArgs(latestArticlesLimit: 2)
        )
        XCTAssertEqual(res.updateArticle.id, created.createArticle.id)
        XCTAssertEqual(res.updateArticle.title, "Updated Title")
    }

    func testDeleteArticle_returnsTrueWhenArticleExists() async throws {
        let created = try await client.mutations.createArticle(
            input: CreateArticleInput(title: "To Delete", authorId: "1"),
            fieldArgs: CreateArticleFieldArgs(latestArticlesLimit: 2)
        )
        let res = try await client.mutations.deleteArticle(id: created.createArticle.id)
        XCTAssertTrue(res.deleteArticle)
    }

    func testDeleteArticle_returnsFalseWhenArticleDoesNotExist() async throws {
        let res = try await client.mutations.deleteArticle(id: "missing-id")
        XCTAssertFalse(res.deleteArticle)
    }

    func testAckPriority_enumMutationArgumentDeserializesReturnsConstantAck() async throws {
        let res = try await client.mutations.ackPriority(level: Priority.high)
        XCTAssertEqual(res.ackPriority, "OK")
    }
}
